runtime/lua/vim/_defaults.lua

```diff
@@ -282,6 +282,21 @@ do
     end,
   })
 
+  vim.api.nvim_create_autocmd('TermOpen', {
+    group = nvim_terminal_augroup,
+    desc = 'Default settings for :terminal buffers',
+    callback = function(args)
+      local buf = args.buf ---@type integer
+      vim.bo[buf].modifiable = false
+      vim.bo[buf].undolevels = -1
+      vim.bo[buf].scrollback = vim.o.scrollback < 0 and 10000 or math.max(1, vim.o.scrollback)
+      vim.bo[buf].textwidth = 0
+      vim.wo[0][buf].wrap = false
+      vim.wo[0][buf].list = false
+      vim.wo[0][buf].winhighlight = 'StatusLine:StatusLineTerm,StatusLineNC:StatusLineTermNC'
```

Comment:
I've only just realized that this will completely override possible user's 'winhighlight'.

It'll probably be better to use something like `vim.opt_local.winhighlight:append()`, but that doesn't support non-current window-buffers. So either using the new `vim._with({ buf = buf }, function() ... end)` or string manipulation like `vim.wo[0][buf].winhighlight = vim.wo[0][buf].winhighlight .. ',...'` should do the trick.